packages/@aws-cdk/aws-certificatemanager/test/test.dns-validated-certificate.ts

```diff
@@ -98,6 +98,23 @@ export = {
     test.done();
   },
 
+  'does not try to validate unresolved tokens'(test: Test) {
+    const stack = new Stack();
+
+    const helloDotComZone = new PublicHostedZone(stack, 'HelloDotCom', {
+      zoneName: Token.asString('hello.com')
+    });
+
+    new DnsValidatedCertificate(stack, 'Cert', {
+      domainName: 'hello.com',
+      hostedZone: helloDotComZone
+    });
+
+    // a bit of a hack: expect(stack) will trigger validation.
```

Comment:
Nice hack with `expect`, but there's a simpler way :). We have a `SynthUtils` class with a `synthesize` method that we use for this kind of situations, so the test becomes:

```ts
  'does not try to validate unresolved tokens'(test: Test) {
    const stack = new Stack();

    const helloDotComZone = new PublicHostedZone(stack, 'HelloDotCom', {
      zoneName: Token.asString('hello.com')
    });

    new DnsValidatedCertificate(stack, 'Cert', {
      domainName: 'hello.com',
      hostedZone: helloDotComZone
    });

    test.doesNotThrow(() => {
      SynthUtils.synthesize(stack);
    });

    test.done();
  },
```

Reply:
Perfect! Ill also update test above this that uses that hack (I didn't come up with it but just went along with it haha).

Do you think it's also worth allowing the `domainName` property in the `DnsValidatedCertificate` to be a token as well? (For me specifically, I only have/need the `zoneName` to be a `Token` but I can see others having the `domainName` in the `DnsValidatedCertificate` be a token.

Reply:
I think it's fine to leave it as it is right now - we can always change it if it's needed.